src/RL/representation/default/heterogeneousstaterepresentation.jl

```diff
@@ -90,6 +91,9 @@ Update the StateRepesentation according to its Type and Featurization.
 """
 function update_representation!(sr::HeterogeneousStateRepresentation, model::CPModel, x::AbstractIntVar)
     update_features!(sr, model)
+    ncon = sr.cplayergraph.numberOfConstraints
+    nvar = sr.cplayergraph.numberOfVariables
+    sr.possibleValuesIdx = map(v -> indexFromCpVertex(sr.cplayergraph, ValueVertex(v)), collect(x.domain)) .- ncon .- nvar
```

Comment:
I definitely need your approval on this line.

Reply:
This line is correct! However, some errors are raised when we compute `valueFeatures` from `possibleValuesIdx` in the case of a `BatchedHeterogeneousTrajectoryState`. I am currently working on it.